backend/pkg/api/data_access/general.go

```diff
@@ -173,3 +177,83 @@ func runQuery[T any](ctx context.Context, db *sqlx.DB, ds *goqu.SelectDataset) (
 
 	return result, nil
 }
+
+// returns the effective balances of the provided validators
+// if onlyActive = true: executed from the vdb premium limits pov, i.e. exited validators account for the EB at exit time
+func (d *DataAccessService) GetValidatorsEffectiveBalances(ctx context.Context, validators []t.VDBValidator, onlyActive bool) (map[t.VDBValidator]uint64, error) {
+	validatorMapping, err := d.services.GetCurrentValidatorMapping()
+	if err != nil {
+		return nil, err
+	}
+
+	// active
+	effectiveBalances := make(map[t.VDBValidator]uint64)
+	var validatorExitEpochs []exp.Expression
+	for _, validator := range validators {
+		if len(validatorMapping.ValidatorMetadata) <= int(validator) {
+			return nil, fmt.Errorf("validator index %d not found in validator mapping", validator)
+		}
+		status := constypes.ValidatorDbStatus(validatorMapping.ValidatorMetadata[validator].Status)
+		if !onlyActive &&
+			(status == constypes.DbSlashed || status == constypes.DbExited) &&
+			validatorMapping.ValidatorMetadata[validator].EffectiveBalance == 0 {
+			// exited & balance withdrawn, need to query latest EB before exit
+			if !validatorMapping.ValidatorMetadata[validator].ExitEpoch.Valid {
+				return nil, fmt.Errorf("validator %d has no exit epoch", validator)
+			}
+			// goqu can't pass tuples directly
+			epochTs := utils.EpochToTime(uint64(validatorMapping.ValidatorMetadata[validator].ExitEpoch.Int64)).Unix()
+			validatorExitEpochs = append(validatorExitEpochs, goqu.L("(fromUnixTimestamp(?), ?)", epochTs, validator))
```

Comment:
Nitpick: should do early continue and have the bulk of the code outside the if, something along these lines:
```suggestion
		plsNameThisVar := !onlyActive &&
			(status == constypes.DbSlashed || status == constypes.DbExited) &&
			validatorMapping.ValidatorMetadata[validator].EffectiveBalance == 0
		if !plsNameThisVar {
			effectiveBalances[validator] = validatorMapping.ValidatorMetadata[validator].EffectiveBalance
			continue
		}
		// exited & balance withdrawn, need to query latest EB before exit
		if !validatorMapping.ValidatorMetadata[validator].ExitEpoch.Valid {
			return nil, fmt.Errorf("validator %d has no exit epoch", validator)
		}
		// goqu can't pass tuples directly
		epochTs := utils.EpochToTime(uint64(validatorMapping.ValidatorMetadata[validator].ExitEpoch.Int64)).Unix()
		validatorExitEpochs = append(validatorExitEpochs, goqu.L("(fromUnixTimestamp(?), ?)", epochTs, validator))
```